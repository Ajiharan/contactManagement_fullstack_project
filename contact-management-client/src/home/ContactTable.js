import React from "react";
import styled from "styled-components";

const ContactTable = ({ contactDatas }) => {
  return (
    <ListContainer>
      <div className="Product__list m-4 p-1">
        {contactDatas.map((res) => (
          <ConactListRow key={res._id}>
            <div className="left">
              <h5>{res.name}</h5>
              <p>
                {res.email}
                {res?.address && `-------[${res.address}]`}
              </p>
              {res?.phoneNo && <p>phone No :- {res.phoneNo}</p>}
            </div>
            <div className="right">
              <button className="btn btn-small btn-danger">select</button>
            </div>
          </ConactListRow>
        ))}
      </div>
    </ListContainer>
  );
};

const ListContainer = styled.div`
  width: 50%;
`;

const ConactListRow = styled.div`
  background-color: rgb(250, 171, 25) !important;
  color: white;
  box-shadow: 5px 5px 10px 0 rgba(0, 0, 0, 0.25),
    -6px -6px 10px 0 rgba(255, 255, 255, 0.3);

  opacity: 0.9;
  font-weight: 600;
  border-radius: 5px;
  padding: 0.5rem;
  margin-top: 1rem;
  display: flex;
  align-items: center;
  .left {
    flex: 1;
  }
`;

export default ContactTable;
