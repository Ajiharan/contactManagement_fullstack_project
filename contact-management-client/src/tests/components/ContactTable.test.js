import React from "react";
import { shallow } from "enzyme";

import ContactTable from "../../home/ContactTable";

test("should test Contact Table as empty props component", () => {
  const wrapper = shallow(<ContactTable contactDatas={[]} loading={false} />);
  expect(wrapper).toMatchSnapshot();
});

test("should test Contact Table as valid props component", () => {
  const wrapper = shallow(
    <ContactTable
      contactDatas={[
        { _id: "bb001245", name: "user", email: "test@gmail.com" },
      ]}
      loading={false}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
