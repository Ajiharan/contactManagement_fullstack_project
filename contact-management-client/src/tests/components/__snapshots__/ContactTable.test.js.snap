// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Contact Table as empty props component 1`] = `
<styled.div>
  <div
    className="Product__list m-4 p-1"
  />
</styled.div>
`;

exports[`should test Contact Table as valid props component 1`] = `
<styled.div>
  <div
    className="Product__list m-4 p-1"
  >
    <Contact
      data={
        Object {
          "_id": "bb001245",
          "email": "test@gmail.com",
          "name": "user",
        }
      }
      key="bb001245"
      setContactData={[Function]}
      setShowModal={[Function]}
    />
  </div>
</styled.div>
`;
